File: leetcode/1545.scala
object Solution {
    def findKthBit(n: Int, k: Int): Char = {
        def S(n :Int): String = {
            def invertBinary(s: String): String = {
                s.map(b => b match {
                    case '0' => '1'
                    case  _  => '0'
                 })
                .mkString
            }
            
            n match {
                case 1 => "0"
                case _ => S(n-1) + "1" + invertBinary(S(n-1).reverse)
            }
        }
        
        S(n)(k-1)
    }
}

File: leetcode/1545.py
class Solution:
    def findKthBit(self, n: int, k: int) -> str:
        invert = lambda s: ''.join(
            '0' if c == '1' else '1' for c in s
        )

        s = '0'
        for _ in range(1, n):
            s = s + '1' + invert(s[::-1])

        return s[k - 1]
